client/graphql/errors.go

```diff
@@ -0,0 +1,118 @@
+package graphql
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"log"
+	"reflect"
+	"strings"
+
+	"github.com/stateful/runme/client/graphql/query"
+)
+
+var ErrNoData = errors.New("no data")
+
+type APIError struct {
+	apiError   error
+	userErrors []query.UserError
+}
+
+func NewAPIError(apiErr error, uErrs ...interface{}) error {
+	e := APIError{}
+	e.SetAPIErrors(apiErr)
+	for _, err := range uErrs {
+		e.SetUserErrors(convertUserErrors(err))
+	}
+	if e.apiError != nil || e.userErrors != nil {
+		return &e
+	}
+	return nil
+}
+
+func (e *APIError) SetAPIErrors(err error) {
+	if err == nil {
+		return
+	}
+	e.apiError = err
+}
+
+func (e *APIError) SetUserErrors(val []query.UserError) {
+	if len(val) == 0 {
+		return
+	}
+	e.userErrors = val
+}
+
+func (e *APIError) Unwrap() error {
+	if e == nil {
+		return nil
+	}
+	return e.apiError
+}
+
+func (e *APIError) Error() string {
+	if e.apiError != nil {
+		return e.apiError.Error()
+	}
+
+	if len(e.userErrors) > 0 {
+		var errs []error
+
+		var b bytes.Buffer
+		if _, err := b.WriteString(userErrorString(e.userErrors[0])); err != nil {
```

Comment:
```suggestion
		b.WriteString(userErrorString(e.userErrors[0]))
```

is fine because `bytes.Buffer.Write()` and related `WriteString()` etc. do not return non-nil error ever. You can check out [implementation](https://github.com/golang/go/blob/1a09d57de58a90987789ef637083aac21533eeb7/src/bytes/buffer.go#L166-L173).